front/lib/models/assistant/actions/data_sources.ts

```diff
@@ -92,7 +92,11 @@ AgentDataSourceConfiguration.init(
         if ((dsConfig.tagsIn === null) !== (dsConfig.tagsNotIn === null)) {
           throw new Error("Tags must be both set or both null");
         }
-        if (dsConfig.tagsMode === "custom") {
+        if (dsConfig.tagsMode === "auto") {
+          if (!dsConfig.tagsIn || !dsConfig.tagsNotIn) {
```

Comment:
It means we must have both values set (eventually empty array) - shouldn't we enforce also for custom ? currently for custom we can have `tagsIn=null` and `tagsNotIn={"xx"}`
Also the error messages does not match ("custom" instead of "auto")

Reply:
for custom we enforce at least one item in one of the array! Does that make sense? 
Fixing the wording thank you 🙏🏻